programming_assignments/tests.py

```diff
@@ -0,0 +1,606 @@
+            if assign_type == "adv":
+                self.assertEqual(
+                    response_data["files_to_be_submitted"],
+                    data["files_to_be_submitted"],
+                )
+                self.assertEqual(response_data["policy"], data["policy"])
+
+    def _create_simple_or_advance_assignment_helper(self, assign_type):
+        """Test: create an assignment.
+
+        Args:
+            assign_type: "simple" or "adv"
+        """
+        # Created by instructor
+        self.login(**ins_cred)
+        self._create_assignment_helper(
+            "Assignment 1",
+            status.HTTP_201_CREATED,
+            assign_type,
+        )
+        self.logout()
+
+        # Created by ta
+        self.login(**ta_cred)
+        self._create_assignment_helper(
+            "Assignment 2",
+            status.HTTP_201_CREATED,
+            assign_type,
+        )
+        self.logout()
+
+        # `HTTP_400_BAD_REQUEST` due to serialization errors
+        self.login(**ins_cred)
+        self._create_assignment_helper(
+            "",
+            status.HTTP_400_BAD_REQUEST,
+            assign_type,
+        )
+        self.logout()
+
+        # `HTTP_401_UNAUTHORIZED` due to `IsInstructorOrTA` permission class
+        self._create_assignment_helper(
+            "Assignment 3",
+            status.HTTP_401_UNAUTHORIZED,
+            assign_type,
+        )
+
+        # `HTTP_403_FORBIDDEN` due to `_is_instructor_or_ta()` method
+        self.login(**stu_cred)
+        self._create_assignment_helper(
+            "Assignment 4",
+            status.HTTP_403_FORBIDDEN,
+            assign_type,
+        )
+        self.logout()
+
+    def test_create_simple_assignment(self):
+        self._create_simple_or_advance_assignment_helper("simple")
+
+    def test_create_advanced_assignment(self):
+        self._create_simple_or_advance_assignment_helper("adv")
+
+    def _list_assignments_helper(self, course_id, user_id, status_code, assign_type):
+        """Helper function for `test_list_assignments()`.
+
+        Args:
+            course_id (int): Course id
+            user_id (int): User id
+            status_code (int): Expected status code of the API call
+            assign_type: "simple" or "adv"
+        """
+        if assign_type == "simple":
+            stud_url_str = (
+                "programming_assignments:simpleprogrammingassignment-"
+                "list-assignments-stud"
+            )
+            stud_url = stud_url_str
+        else:
+            stud_url_str = (
+                "programming_assignments:advancedprogrammingassignment-"
+                "list-assignments-stud"
+            )
+            stud_url = stud_url_str
+
+        if user_id != 3 and assign_type == "simple":
+            url = reverse(
+                "programming_assignments:simpleprogrammingassignment-list-assignments",
+                args=[course_id],
```

Comment:
take reverse outside of if else for all

Reply:
if simeple:
    cases for stud and inst
elif advance
    cases for stud and inst